### ISSUE TITLE
 No gradients provided for any variable, check your graph for ops that do not support gradients, between variables

### ISSUE THREAD

**Issue by anshumitts** [vineetm/ell-881-2018-deep-learning#22]
```
opt = tf.train.GradientDescentOptimizer(learning_rate=0.02)

with tf.Session() as sess:
    sess.run(tf.global_variables_initializer())
    sess.run(train_init_op)
    for i in range(10):
        batch_x, batch_y = sess.run(next_batch)
        logits = sess.run(fc8,feed_dict={inp:batch_x,keep_prob:0.5})
        loss = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(labels=batch_y,logits=logits))
        opt.minimize(loss,var_list=tf.trainable_variables())
        
```

**Comment by vineetm** [comment]
@anshumitts Sorry for replying late. A few questions:

1. Where are you computing gradients in this function? Isn't that what the system is complaining about?
2. Are you intentionally using **non-eager** mode?

**Comment by anshumitts** [comment]
@vineetm  Yes, This is intentional. As this was the class work.
I don't know how else were we supposed to do this.
